core/src/main/scala/besom/internal/codecs.scala

```diff
@@ -486,175 +485,64 @@ trait DecoderHelpers:
-      decodeAsPossibleSecret(value, label).flatMap { odv =>
-        odv
-          .traverseValidatedResult { v =>
-            aDecoder
-              .decode(v, label)
-              .orElse(bDecoder.decode(v, label))
-              .orElse(cDecoder.decode(v, label))
-              .orElse(dDecoder.decode(v, label))
-              .orElse(eDecoder.decode(v, label))
-              .orElse(fDecoder.decode(v, label))
-              .orElse(gDecoder.decode(v, label))
-              .asInstanceOf[ValidatedResult[DecodingError, OutputData[A | B | C | D | E | F | G]]]
-          }
-          .map(_.flatten)
-      }
+    override def mapping(value: Value, label: Label): Validated[DecodingError, A | B] = ???
 
-    override def mapping(value: Value, label: Label): Validated[DecodingError, A | B | C | D | E | F | G] = mappingThrowingPlug
-  end unionDecoder7
-
-  def unionDecoder8[A, B, C, D, E, F, G, H](using
-    aDecoder: Decoder[A],
-    bDecoder: Decoder[B],
-    cDecoder: Decoder[C],
-    dDecoder: Decoder[D],
-    eDecoder: Decoder[E],
-    fDecoder: Decoder[F],
-    gDecoder: Decoder[G],
-    hDecoder: Decoder[H]
-  ): Decoder[A | B | C | D | E | F | G | H] = new Decoder[A | B | C | D | E | F | G | H]:
-    override def decode(value: Value, label: Label): ValidatedResult[DecodingError, OutputData[A | B | C | D | E | F | G | H]] =
-      decodeAsPossibleSecret(value, label).flatMap { odv =>
-        odv
-          .traverseValidatedResult { v =>
-            aDecoder
-              .decode(v, label)
-              .orElse(bDecoder.decode(v, label))
-              .orElse(cDecoder.decode(v, label))
-              .orElse(dDecoder.decode(v, label))
-              .orElse(eDecoder.decode(v, label))
-              .orElse(fDecoder.decode(v, label))
-              .orElse(gDecoder.decode(v, label))
-              .orElse(hDecoder.decode(v, label))
-              .asInstanceOf[ValidatedResult[DecodingError, OutputData[A | B | C | D | E | F | G | H]]]
-          }
-          .map(_.flatten)
-      }
-
-    override def mapping(value: Value, label: Label): Validated[DecodingError, A | B | C | D | E | F | G | H] = mappingThrowingPlug
-  end unionDecoder8
+  inline def error(msg: String, label: Label, cause: Throwable = null): DecodingError = DecodingError(msg, cause, label)
 
   inline def derived[A](using m: Mirror.Of[A]): Decoder[A] =
     lazy val labels           = CodecMacros.summonLabels[m.MirroredElemLabels]
     lazy val instances        = CodecMacros.summonDecoders[m.MirroredElemTypes]
     lazy val nameDecoderPairs = labels.map(NameUnmangler.unmanglePropertyName).zip(instances)
 
     inline m match
-      case s: Mirror.SumOf[A] => decoderSum(s, nameDecoderPairs)
-      case p: Mirror.ProductOf[A] =>
-        decoderProduct(p, nameDecoderPairs)
+      case s: Mirror.SumOf[A]     => decoderSum(s, nameDecoderPairs)
+      case p: Mirror.ProductOf[A] => decoderProduct(p, nameDecoderPairs)
+  end derived
+
+  inline def discriminatory[A](
```

Comment:
I don't think this should be inline. Why is it inline? Is this code somehow specific and has to be inlined in the package code? If not, it just bloats the package and can be safely non-inline method to be called in core sdk.

Also, nitpick: I'd name this probably `discriminated` and below `nonDiscriminated`.